frontend/src/utils/cookies.ts

```diff
@@ -0,0 +1,64 @@
+import {IAPIAuthResponse} from 'src/interfaces/IAPI';
+
+/**
+ * Set a cookie in the browser with the
+ * given name, value, and expiration date.
+  * @param {string} cookieName - Cookie name
+  * @param {string} cookieValue - Cookie value
+  * @param {number} daysToExpire - Number of days to expire the cookie
+ */
+export function setCookie(
+    cookieName: string,
+    cookieValue: string,
+    daysToExpire: number,
+): void {
+  const date = new Date();
+  date.setTime(date.getTime() + (daysToExpire * 24 * 60 * 60 * 1000));
+  const expires = 'expires=' + date.toUTCString();
+  document.cookie = cookieName + '=' + cookieValue + ';' + expires + ';path=/';
+}
+
+/**
+  * Get a cookie in the browser from a given name.
+  * @param {string} cookieName - Cookie name
+  * @return {string} - Cookie value
+ */
+export function getCookie(cookieName: string): string {
```

Comment:
Is there any issue working with cookies locally? Especially with auth features.

Reply:
Working on this cookie can be a security issue as those cookies are not secure and are not reliable. 
In a production application, cookies should not be manipulated like this in the frontend.

Firstly, the `HttpOnly`, `Secure`, `SameSite` are not set. Those flags should be set as :

=> `HttpOnly` prevent access of cookie via Javascript.
=> `SameSite`: the cookie affected by this instruction will only be sent if the request comes from the same website.
=> `Secure`: only send the cookie via secured HTTP channel. 

Cookies might also be intercepted by a third party and altered: 
    * In a man-in-the-middle (MITM) attacks (by unsecured HTTP channel as `Secure` is not set)
    * in Cross-Site Scripting (XSS) attacks (access from Javascript)
    * From browser memory, the cookie is not encrypted.

Also, the cookie is set to expire in 60 days, in this application. It is too long. 2 days, in my opinion, are enough.

That’s why cookies implemented this way are a security issue.

To mitigate this problem, some solutions to fix those security issues include:
* Secure the cookie with the flags mentioned above
* Implements `accessToken` and `refreshToken` logic in the application (See https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them)
* Not storing and reading sensible pieces of information like username from the cookies as they can be altered but only store a hashed `refreshToken`

These is the issues in my opinion. Is there something else worth mentioning?
If you want, I can implement what I explained and fix the security issues in the pull request.

Thanks!

Reply:
Thanks for the reply, that's enough for our evaluation :).